### PR TITLE
[WIP][2.4][Form] Add option "widget" to ChoiceType

### PR DESCRIPTION
**Pull request by bamarni**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | no |
| Fixed tickets | #6602 |
| License | MIT |
| Doc PR | not yet |
- [ ] add / fix tests / CS / PHPDocs
- [ ] submit changes to the documentation
- [ ] update php templates and twig table layout

I've implemented the behavior described in #6602.
### Choice
- Also, does `choice_widget_select` now fit better than `choice_widget_collapsed`?
### Doctrine
- For the original entity_id use case, it'd simply mean setting the `property` option to `id`.

``` php
<?php

$builder->add('field', 'entity', array(
    'class' => 'MyBundle:Product',
    'widget' => 'text',
    'property' => 'id',
));
```

Then the choice list won't be loaded in advance but validated on the fly.

What do you think about the approach?  cc @bschussek

### CODE REVIEW

**Comment by colinodell** [comment]
Is this feature still being worked on?

**Comment by bamarni** [comment]
Not since a while, I was still waiting for @webmozart feedback about the approach. Basically it does a bit more than the suggested "entity_id" form type would, in the sense that it also allow to use any other entity property (of course it has to be unique). I can continue this PR if we decide this approach is the way to go.

**Comment by webmozart** [comment]
Hi @bamarni, thank you again for working on this issue! Sorry you had to wait so long for my feedback.

The ChoiceList implementation changed quite a bit in Symfony 2.7. Do you think you could rebase your work on the 2.8 branch?

**Comment by bamarni** [comment]
@webmozart : thanks for the feedback, I've implemented the first part for the moment (on the Form component), cf. #15053

File: src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php [hunk 1]
```diff
@@ -45,7 +47,9 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             throw new LogicException('Either the option "choices" or "choice_list" must be set.');
         }
 
-        if ($options['expanded']) {
+        if (in_array($options['widget'], array('radio', 'checkbox'))) {
```
Comment (bamarni) [inline src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php]: now I get your comment! what about an "expanded" widget? see #15053

**Pull request closed without merging**